go.mod

```diff
@@ -1,136 +1,134 @@
 module github.com/GoogleCloudPlatform/prometheus-engine
 
-go 1.17
+go 1.16
```

Comment:
I had to bump this in the past because otherwise something about Go modules kept failing.
Now I could move it back to the Go version also defined in the Prometheus version we depend on. It's just the minimum required Go version, so this will allow building from more environments.

Reply:
Ok - do we need to downgrade the docker image use to build our binaries to [`golang:1.16-buster`](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/a7e0126d79a63715f92e63e7d5103462d42f98b8/cmd/frontend/Dockerfile#L2)?

Reply:
No, it's just the minimum required version. We can build with anything higher.